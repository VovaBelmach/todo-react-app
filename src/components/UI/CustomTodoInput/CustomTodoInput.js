import React from "react";
import PropTypes from "prop-types";
import Checkbox from "../Checkbox/Checkbox";
import TextInput from "../TextInput/TextInput";

const CustomTodoInput = (props) => {
  return (
    <>
      <Checkbox isDisabled={true} />
      <TextInput
        name={props.name}
        value={props.value}
        placeholder={props.placeholder}
        onChangeHandler={props.onChangeHandler}
        onBlurHandler={props.onBlurHandler}
      />
    </>
  );
};

CustomTodoInput.propTypes = {
  name: PropTypes.string,
  value: PropTypes.any,
  placeholder: PropTypes.string,
  onChangeHandler: PropTypes.func,
  onBlurHandler: PropTypes.func,
};

export default CustomTodoInput;
