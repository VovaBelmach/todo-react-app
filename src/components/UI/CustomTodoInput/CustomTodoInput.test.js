import React from "react";
import renderer from "react-test-renderer";
import CustomTodoInput from "./CustomTodoInput";

describe("CustomTodoInput", () => {
  it("should renders correctly", () => {
    const tree = renderer.create(<CustomTodoInput name="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
