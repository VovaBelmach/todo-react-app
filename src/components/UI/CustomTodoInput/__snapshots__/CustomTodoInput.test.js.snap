// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTodoInput should renders correctly 1`] = `
Array [
  <label
    className=""
  >
    <input
      className="checkbox"
      disabled={true}
      type="checkbox"
    />
  </label>,
  <input
    className="textInput"
    name="test"
    type="text"
  />,
]
`;
